tests/foreman/api/test_registration.py

```diff
@@ -0,0 +1,71 @@
+"""Tests for registration.
+
+:Requirement: Registration
+
+:CaseLevel: Acceptance
+
+:CaseComponent: Registration
+
+:CaseAutomation: Automated
+
+:CaseImportance: Critical
+
+:Assignee: sbible
+
+:TestType: Functional
+
+:Upstream: No
+"""
+import pytest
+
+pytestmark = pytest.mark.tier1
+
+
+@pytest.mark.e2e
+@pytest.mark.no_containers
+@pytest.mark.rhel_ver_match('[^6].*')
+def test_host_registration_end_to_end(
+    module_org,
+    module_location,
+    module_ak_with_synced_repo,
+    module_target_sat,
+    module_capsule_configured,
+    rhel_contenthost,
+):
+    """Verify content host registration with global registration
+
+    :id: 219567a8-856a-11ed-944d-03d9b43011c2
+
+    :steps:
+        1. Register host with global registration template to Satellite and Capsule
+
+    :expectedresults: Host registered successfully
+    """
+    command = module_target_sat.api.RegistrationCommand(
+        organization=module_org,
+        activation_keys=[module_ak_with_synced_repo.name],
+        location=module_location,
+    ).create()
+    result = rhel_contenthost.execute(command)
+    assert result.status == 0, f'Failed to register host: {result.stderr}'
+
+    # Update module_capsule_configured to include module_org/module_location
+    nailgun_capsule = module_target_sat.api.SmartProxy().search(
+        query={'search': f'name={module_capsule_configured.hostname}'}
+    )[0]
```

Comment:
non-blocking, I believe this could be a `Capsule`'s property.

```python
@property
def nailgun_smart_proxy(self):
    return self.satellite.api.SmartProxy().search(query={'search': f'name={self.hostname}'})[0]
```

```suggestion
    nailgun_capsule = module_capsule_configured.nailgun_smart_proxy
```

This is just a suggestion, I did not test it.

Reply:
Added and remove redundant `internal_capsule` property from satellite class

Reply:
ah I missed this, updated